tests/kafkatest/tests/connect/connect_distributed_test.py

```diff
@@ -519,6 +535,105 @@ def test_bounce(self, clean, connect_protocol):
 
         assert success, "Found validation errors:\n" + "\n  ".join(errors)
 
+    @cluster(num_nodes=6)
+    @matrix(clean=[True, False], connect_protocol=['sessioned', 'compatible', 'eager'])
+    def test_exactly_once_source(self, clean, connect_protocol):
+        """
+        Validates that source and sink tasks that run continuously and produce a predictable sequence of messages
+        run correctly and deliver messages exactly once when Kafka Connect workers undergo bounces, both clean and unclean.
```

Comment:
Could you also help update the description in `test_bounce`? It tests not only `clean` bounces, right?

Reply:
Yep, done.